### DISCUSSION TITLE
My dream - search both messages and **changed codes** in full commit history

### DISCUSSION THREAD

**Discussion by edwinyzh** [community/community#9835] [Other Feature Feedback, Questions, & Ideas]
First, my kudos and congratulations to the github code search implementation!

Secondly, by looking at the instructions at the bottom of the cs.github front page, it seems that no commits search support?

Ideally, I'd like to **search for changed/removed codes in the entire repo history**. For example, let's say there was function call like `function1(...)` in the source code, but it's been removed sometime in the past, so if you search in the current repo you won't find anything.
However, I want to be able to search for `function1` and the code search will tell me 2 months ago there used to be a function call like that.

Hope it makes sense.

**Comment by megoels** [comment]
Hi, Is there any plan to add commits to GitHub code search ?

**Comment by mbaas2** [comment]
Totally agree - please add that!
